### PR TITLE
Add Stable Diffusion example

### PR DESCRIPTION
**Pull request by tsailiming**
## Description
This PR creates an example of using stable diffusion fine-tuning on a Ray cluster and serving

## How Has This Been Tested?
- This has been tested on OpenShift AI 2.10 running on OpenShift 4.15 on a AWS environment with a single p3.8xlarge instance.
- Provision a OCP 4.15 cluster on AWS with 3 ODF worker nodes and 1 p3.8xlarge worker
- Deoloy OpenShift AI using make deploy-oai
- Run make setup-ray-distributed-training
- After deploying the example, run through the 1 to 4 notebooks.

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: examples/stable-diffusion-dreambooth/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+## Running the Examples
+
+In the workbench, navigate to `examples/stable-diffusion-dreambooth/notebook` to begin exploring the notebooks.
+
+### Fine-tune of Stable Diffusion with DreamBooth and Ray Train
+
+The `01 raytrain-stablediffusion.ipynb` notebook provides various steps from the Ray docs [example](https://docs.ray.io/en/latest/train/examples/pytorch/dreambooth_finetuning.html#step-5-generate-images-of-the-subject) and will create a Ray cluster with 4 workers with 1 GPU each to run the jobs.
+
```
Comment (KPostOffice) [inline examples/stable-diffusion-dreambooth/README.md:87]: Not sure what the cause is exactly but some of the logs from training look strange. It works fine and is fairly readable just oddly spread out in some places with `[A` characters everywhere.
image

File: examples/stable-diffusion-dreambooth/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+```    
+
+The images will be shown using `matplotlib`.
+
+![diog diog pictures](docs/diog-diog.png)
+
+### Uploading and Serving the Model Using KServe
+
+* `02 raytrain-stablediffusion-upload-model.ipynb` uploads the tuned model to a local S3 bucket.
+* `03 raytrain-stablediffusion-model-serving.ipynb` sets up KServe with PyTorch
```
Comment (KPostOffice) [inline examples/stable-diffusion-dreambooth/README.md:112]: Inference never came up for me, I was having trouble with the Service Mesh operator. Does it rely on this? If so I think the issue can likely be ignored.

**Review by KPostOffice — COMMENTED** [review]
Looks good to me overall. A couple comments. I think I'll try with an actual fresh cluster tomorrow

File: examples/stable-diffusion-dreambooth/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+> [!WARNING]  
+> The setup script will delete all clusterqueues and resourceflavors in the cluster.
+
+``` bash
+make setup-ray-distributed-training
+```
+
+## Running the Examples
+
+In the workbench, navigate to `examples/stable-diffusion-dreambooth/notebook` to begin exploring the notebooks.
```
[the remark this answers is not in the document]
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/README.md:82]: I thought the directory is `examples/stable-diffusion-dreambooth/notebook`?

File: examples/stable-diffusion-dreambooth/Makefile [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		sleep 10; \
+	done
+
+	oc apply -f $(BASE)/yaml/operators/odf-storagecluster.yaml
+	oc apply -f $(BASE)/yaml/operators/odf-storagesystem.yaml
+	
+	@until oc get storagecluster ocs-storagecluster -n openshift-storage -o jsonpath='{.status.conditions[?(@.type=="Available")].status}' | grep -q "True"; do \
+		echo "Waiting for ODF to be ready..."; \
+		sleep 60; \
+	done
```
[the remark this answers is not in the document]
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/Makefile]: I have enhanced the check in the latest commit. Might want to check whether is because there are more than 3 `cluster.ocs.openshift.io/openshift-storage=` labels? I had ran this in a new clean system and was ok for me.

File: examples/stable-diffusion-dreambooth/README.md (hunk 2 above)
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/README.md:112]: Yes. It is a prereq. The latest commit will check for the Service Mesh operator is ready before installing OpenShift AI operator.

File: examples/stable-diffusion-dreambooth/README.md (hunk 1 above)
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/README.md:87]: Yeah. It think that's the output rendering problem. Not sure why this is happening.

File: examples/stable-diffusion-dreambooth/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/README.md:82]: I saw the typo in the workbench. Will correct this.

File: examples/stable-diffusion-dreambooth/notebook/raycluster.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
+apiVersion: ray.io/v1
+kind: RayCluster
+metadata:
+  name: stable-diffusion
```
[the remark this answers is not in the document]
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/notebook/raycluster.yaml:4]: Fixed this by using a different workbench name.

File: examples/stable-diffusion-dreambooth/README.md (hunk 2 above)
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/README.md:118]: I have enhanced the Makefile to check for the operator readiness first. Took a clean run and is working fine for me.

File: examples/stable-diffusion-dreambooth/Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (KPostOffice) [reply examples/stable-diffusion-dreambooth/Makefile]: Not sure what the issue is, I'm still having the same issue on a fresh cluster. I have only 3 worker nodes. I'm on a ROSA cluster. I'm not sure if that matters at all.

**Review by KPostOffice — COMMENTED** [review]
Everything worked great! I had some issues with the final two steps due to ssl verification errors.

File: examples/stable-diffusion-dreambooth/README.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+  You can install OpenShift Data Foundation (ODF) to achieve this. ODF will require a minimum of 3 worker nodes to start with. 
+
+  ```
+  make deploy-odf
+  ```
+
+  The CSI plugin daemonset will have to be updated to tolerate the GPU taint. 
+
+  ```
```
[the remark this answers is not in the document]
Comment (tsailiming) [reply examples/stable-diffusion-dreambooth/README.md]: Fixed

**Comment by tsailiming** [comment]
> Everything worked great! I had some issues with the final two steps due to ssl verification errors, can the app be changed to `verify=False` when submitting requests to the inference backend? (same with the request in examples/stable-diffusion-dreambooth/notebook/model/torchserve-query.py)

Just fixed this in `torchserve-query.py` and for the streamlit `app.py`

**Review by KPostOffice — APPROVED** [review]
/lgtm
